cidc_api/models.py

```diff
@@ -268,32 +284,30 @@ def patch_manifest(trial_id: str, manifest_patch: dict, session: Session):
 
             TODO: apply this update directly to the not-yet-existent TrialMetadata.assays field
         """
-        TrialMetadata.patch_trial_metadata(trial_id, manifest_patch, session=session)
+        trial = TrialMetadata.select_for_update_by_trial_id(trial_id)
```

Comment:
since this code is repeated verbatim above, maybe we should make a function that wraps this:
```python
@staticmethod
@with_default_session
def _select_and_patch_trial_metadata(trial_id: str, json_patch: dict, session: Session):
    trial = TrialMetadata.select_for_update_by_trial_id(trial_id)
    trial._patch_trial_metadata(assay_patch, session=session)
```
We still get the benefits of row-level locking on `trial` (as I understand it).